journal/src/main/java/io/zeebe/journal/file/FrameUtil.java

```diff
@@ -49,14 +49,22 @@ public static void markAsIgnored(final ByteBuffer buffer, final int offset) {
       if (val != IGNORE) {
         return Optional.of((int) val);
       }
-    } catch (BufferUnderflowException e) {
+    } catch (final BufferUnderflowException e) {
       // nothing to read - reset
     }
 
     buffer.reset();
     return Optional.empty();
   }
 
+  public static boolean hasValidVersion(final ByteBuffer buffer, final int position) {
+    if (buffer.capacity() < position + LENGTH) {
+      return false;
+    }
+    final byte val = buffer.get(position);
+    return val != IGNORE;
+  }
```

Comment:
Why not use the existing code that already reads the version?

```suggestion
  public static boolean hasValidVersion(final ByteBuffer buffer, final int position) {
    final boolean valid = readVersion(buffer).isPresent();
    buffer.reset();
    return valid;
  }
```